### PR TITLE
Hide bot token in errors

### PR DESCRIPTION
**Pull request by WaffleLapkin**
This fixes a potential[^1] security vulnerability -- if bot shows errors from teloxide to the user & for some reason network error happened[^2] the url of the request would be included in the error. Since TBA includes bot token in the error this may lead to token leakage.

This PR fixes that issue by removing the token from the urls of `reqwest::Error`, we try to redact only the token, but if we fail we remove the whole url.

This can be tested by using a very low timeout value for the http reqwest client:
```rust
let client = reqwest::Client::builder()
    .timeout(std::time::Duration::from_millis(1))
    .build()
    .unwrap();

let bot = Bot::from_env_with_client(client).auto_send();

// see if the token is redacted when network error (timeout) happens
// while sending common requests
let _ = dbg!(bot.get_me().await);

// see if the token is redacted when network error (timeout) happens
// while downloading files ("path" is unimportant as the timeout is so
// low the request probably won't even be sent)
let _ = dbg!(bot.download_file_stream("path").next().await);
```

For me this gives the following result:
```text
[t.rs:26] bot.get_me().await = Err(
    Network(
        reqwest::Error {
            kind: Request,
            url: Url {
                scheme: "https",
                cannot_be_a_base: false,
                username: "",
                password: None,
                host: Some(
                    Domain(
                        "api.telegram.org",
                    ),
                ),
                port: None,
                path: "/token:redacted/GetMe",
                query: None,
                fragment: None,
            },
            source: TimedOut,
        },
    ),
)
[t.rs:31] bot.download_file_stream("path").next().await = Some(
    Err(
        reqwest::Error {
            kind: Request,
            url: Url {
                scheme: "https",
                cannot_be_a_base: false,
                username: "",
                password: None,
                host: Some(
                    Domain(
                        "api.telegram.org",
                    ),
                ),
                port: None,
                path: "/file/token:redacted/path",
                query: None,
                fragment: None,
            },
            source: TimedOut,
        },
    ),
)
```

Note that this branch is created from `v0.4.4` tag and not `master,` since the latter has breaking changes and we want to release this fix as `v0.4.5`. We'll need to make a release from this branch directly.

[^1]: Note that there are recorded cases where the token got exposed.
[^2]: Note that this can theoretically be controlled by the user when
      sending/downloading bigger files.

### CODE REVIEW

**Comment by WaffleLapkin** [comment]
It seems like the timeline of events should be as follows:
1. @Hirrolot reviews this PR
2. I release a new `teloxide-core` version
3. We resolve conflicts with master (by merging master to this branch)
4. We merge this PR

File: src/errors.rs [hunk 1]
```diff
@@ -727,3 +728,54 @@ pub enum ApiError {
...
+        // Usually the url looks like "bot<token>/..." or "file/bot<token>/...".
+        let (beginning, segment) = match segments.next() {
+            Some("file") => ("file/", segments.next()),
+            segment => ("", segment),
+        };
+
+        if let Some(token) = segment.and_then(|s| s.strip_prefix("bot")) {
+            // make sure that what we are about to delete looks like a bot token
+            if let Some((id, secret)) = token.split_once(':') {
+                if secret.len() >= 35
```
Comment (Hirrolot) [inline src/errors.rs:761]: What is 35 here?

Comment (WaffleLapkin) [reply src/errors.rs:761]: The lenght of the secret. On all bot tokens I could find it's 35

Comment (Hirrolot) [reply src/errors.rs:761]: Ok, then consider giving the magic number a name: `const SECRET_LENGTH: usize = 32;` or something like this.

Comment (WaffleLapkin) [reply src/errors.rs]: Done

**Review by Hirrolot — APPROVED** [review]

**Pull request merged**